pkgs/build-support/trivial-builders.nix

```diff
@@ -300,6 +300,13 @@ rec {
    *     |       `-- stack.fish -> /nix/store/6lzdpxshx78281vy056lbk553ijsdr44-stack-2.1.3.1/share/fish/vendor_completions.d/stack.fish
    * ...
    *
+
+   * # creates multiple outputs, where the `out` output is the result
+   * # of combining the main outputs of all packages in `paths`, and
+   * # `bin` and `dev` are the result of combining their corresponding
+   * # outputs of all packages in `paths`
+   * symlinkJoin { name = "multiexample"; paths = [ pkgs.libpng pkgs.openssl ]; outputs = [ "out" "bin" "dev" ]; }
```

Comment:
Thinking about this more, this will probably only be useful if the secondary outputs are disjoint with the set of main outputs of the paths. Otherwise, you will end up with the same content in multiple outputs like here with `openssl.bin`.

Reply:
Yes, indeed. I don't think there's a "right answer" here, but I think the current one should cover most use-cases. In this case, you can ignore the `bin` output, since it's more likely you want the main output of all paths, but if you really have a need for it, it's available. Or do you have a better idea of how this should work?

Reply:
Yeah, I do not think there is right answer either. But we probably should not include things like this in an example – I would either remove the `bin` output or use `openssl.out`. And add a comment emphasizing that behaviour.

Reply:
I've updated the example a bit, removing the `bin` output and adding `man` instead. I also added a comment about explicitly using `<pkg>.out` for packages where it's not the main output. I guess an alternative to this would be to call our main output something different, like `all_mains`, and let the others be pure summations of their respective outputs.

Reply:
Hmm, actually that raises another question. If one explicitly adds `openssl.out` should `openssl.dev` (actually `openssl.out.dev`) still end up in dev? Or should we use `lib.optionalString ((path.outputUnspecified or false) && path ? ${output})`.

Reply:
Hm, yeah, that's a good question. I'm leaning towards the latter, but I'm not really sure..